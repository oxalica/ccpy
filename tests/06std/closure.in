def f(x=0):
    def g():
        nonlocal x
        x = x + 1
        return x
    return g

a, b, c = f(), f(), f(10)
print(a(), a(), a())
print(b(), b())
print(c(), c())
