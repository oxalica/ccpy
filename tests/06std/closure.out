Std done

1 2 3
1 2
11 12
Done
